events/serializers.py

```diff
@@ -0,0 +1,1760 @@
+                "name",
+                "last_modified_time",
+                "deleted",
+                "replaced_by",
+            ]
+            for key in ret.keys() - keys_to_preserve:
+                del ret[key]
+            ret["name"] = utils.get_deleted_object_name()
+            return ret
+
+        # Remove personal information fields from public API
+        user = self.context["request"].user
+        if (
+            not settings.ENABLE_EXTERNAL_USER_EVENTS
+            or not (
+                user.is_authenticated
+                and (
+                    user.is_regular_user_of(obj.publisher)
+                    or user.is_admin_of(obj.publisher)
+                )
+            )
+            and obj.created_by != user
+        ):
+            for field in self.personal_information_fields:
+                if field in ret:
+                    del ret[field]
+
+        if self.context["request"].accepted_renderer.format == "docx":
+            ret["end_time_obj"] = obj.end_time
+            ret["start_time_obj"] = obj.start_time
+            ret["location"] = obj.location
+
+        if obj.start_time and not obj.has_start_time:
+            # Return only the date part
+            ret["start_time"] = obj.start_time.astimezone(LOCAL_TZ).strftime("%Y-%m-%d")
+        if obj.end_time and not obj.has_end_time:
+            # If the event is short (<24h), then no need for end time
+            if obj.start_time and obj.end_time - obj.start_time <= timedelta(days=1):
+                ret["end_time"] = None
+
+            else:
+                # If we're storing only the date part, do not pretend we have the exact time.
+                # Timestamp is of the form %Y-%m-%dT00:00:00, so we report the previous date.
+                ret["end_time"] = utils.start_of_previous_day(obj.end_time).strftime(
+                    "%Y-%m-%d"
+                )
+
+        del ret["has_start_time"]
+        del ret["has_end_time"]
+        if hasattr(obj, "days_left"):
+            ret["days_left"] = int(obj.days_left)
+        if self.skip_empties:
+            for k in list(ret.keys()):
+                val = ret[k]
+                try:
+                    if val is None or len(val) == 0:
+                        del ret[k]
+                except TypeError:
+                    # not list/dict
+                    pass
+        request = self.context.get("request")
+        if request and not request.user.is_authenticated:
+            del ret["publication_status"]
+
+        return ret
+
+
+class EventSerializerV0_1(EventSerializer):  # noqa: N801
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("context", {}).setdefault("include", []).append("image")
+        super().__init__(*args, **kwargs)
+
+    def to_representation(self, obj):
+        ret = super().to_representation(obj)
+        _format_images_v0_1(ret)
+        return ret
+
+
+# RegistrationSerializer is in this file to avoid circular imports
```

Comment:
Is this still relevant?

Reply:
Not that much. I moved the serializer to `registrations.serializers` and combined it with `RegistrationBaseSerializer`.